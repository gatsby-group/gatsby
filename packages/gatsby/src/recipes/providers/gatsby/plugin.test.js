const path = require(`path`)

const plugin = require(`./plugin`)

const root = path.join(__dirname, `./fixtures`)
const name = `gatsby-plugin-foo`

describe(`gatsby-config`, () => {
  test(`adds a plugin to a gatsby config`, async () => {
    await plugin.create({ root }, { name })

    const result = await plugin.read({ root })

    expect(result).toContain(`gatsby-plugin-foo`)

    await plugin.destroy({ root }, { name })
  })

  test(`retrieves plugins from a config`, async () => {
    const result = await plugin.read({ root })

    expect(result).toEqual([
      `gatsby-source-filesystem`,
      `gatsby-transformer-sharp`,
      `gatsby-plugin-emotion`,
      `gatsby-plugin-typography`,
      `gatsby-transformer-remark`,
      `gatsby-plugin-sharp`,
      `gatsby-plugin-google-analytics`,
      `gatsby-plugin-manifest`,
      `gatsby-plugin-offline`,
      `gatsby-plugin-react-helmet`,
    ])
  })
})
