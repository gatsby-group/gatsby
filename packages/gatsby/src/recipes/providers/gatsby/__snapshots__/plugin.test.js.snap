// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin resource all returns an array of plugins 1`] = `
Array [
  Object {
    "id": "gatsby-source-filesystem",
    "name": "gatsby-source-filesystem",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-transformer-sharp",
    "name": "gatsby-transformer-sharp",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-emotion",
    "name": "gatsby-plugin-emotion",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-typography",
    "name": "gatsby-plugin-typography",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-transformer-remark",
    "name": "gatsby-transformer-remark",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-sharp",
    "name": "gatsby-plugin-sharp",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-google-analytics",
    "name": "gatsby-plugin-google-analytics",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-manifest",
    "name": "gatsby-plugin-manifest",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-offline",
    "name": "gatsby-plugin-offline",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
  Object {
    "id": "gatsby-plugin-react-helmet",
    "name": "gatsby-plugin-react-helmet",
    "shadowableFiles": Array [],
    "shadowedFiles": Array [],
  },
]
`;

exports[`gatsby-plugin resource e2e plugin resource test with hello world starter: GatsbyPlugin create 1`] = `
Object {
  "_message": "Installed gatsby-plugin-foo in gatsby-config.js",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test with hello world starter: GatsbyPlugin create plan 1`] = `
Object {
  "currentState": "/**
 * Configure your Gatsby site with this file.
 *
 * See: https://www.gatsbyjs.org/docs/gatsby-config/
 */
module.exports = {
  /* Your site config here */
  plugins: []
};",
  "describe": "Install gatsby-plugin-foo in gatsby-config.js",
  "diff": "[31m- Original  - 2[39m
[32m+ Modified  + 3[39m

[33m@@ -5,5 +5,6 @@[39m
[2m   */[22m
[2m  module.exports = {[22m
[2m    /* Your site config here */[22m
[31m-   plugins: [][39m
[31m- };[39m
[32m+   plugins: [\\"gatsby-plugin-foo\\"],[39m
[32m+ }[39m
[32m+[39m",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "newState": "/**
 * Configure your Gatsby site with this file.
 *
 * See: https://www.gatsbyjs.org/docs/gatsby-config/
 */
module.exports = {
  /* Your site config here */
  plugins: [\\"gatsby-plugin-foo\\"],
}
",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test with hello world starter: GatsbyPlugin destroy 1`] = `undefined`;

exports[`gatsby-plugin resource e2e plugin resource test with hello world starter: GatsbyPlugin update 1`] = `
Object {
  "_message": "Installed gatsby-plugin-foo in gatsby-config.js",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test with hello world starter: GatsbyPlugin update plan 1`] = `
Object {
  "currentState": "/**
 * Configure your Gatsby site with this file.
 *
 * See: https://www.gatsbyjs.org/docs/gatsby-config/
 */
module.exports = {
  /* Your site config here */
  plugins: [\\"gatsby-plugin-foo\\"],
}
",
  "describe": "Install gatsby-plugin-foo in gatsby-config.js",
  "diff": "[2mCompared values have no visual difference.[22m",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "newState": "/**
 * Configure your Gatsby site with this file.
 *
 * See: https://www.gatsbyjs.org/docs/gatsby-config/
 */
module.exports = {
  /* Your site config here */
  plugins: [\\"gatsby-plugin-foo\\"],
}
",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin create 1`] = `
Object {
  "_message": "Installed gatsby-plugin-foo in gatsby-config.js",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin create plan 1`] = `
Object {
  "currentState": "const redish = \`#c5484d\`;
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`
  },
  plugins: [{
    resolve: \`gatsby-source-filesystem\`,
    options: {
      path: \`\${__dirname}/src/pages\`,
      name: \`pages\`
    }
  }, \`gatsby-transformer-sharp\`, \`gatsby-plugin-emotion\`, {
    resolve: \`gatsby-plugin-typography\`,
    options: {
      pathToConfigModule: \`src/utils/typography\`
    }
  }, {
    resolve: \`gatsby-transformer-remark\`,
    options: {
      plugins: [{
        resolve: \`gatsby-remark-images\`,
        options: {
          maxWidth: 590
        }
      }, {
        resolve: \`gatsby-remark-responsive-iframe\`,
        options: {
          wrapperStyle: \`margin-bottom: 1.0725rem\`
        }
      }, \`gatsby-remark-prismjs\`, \`gatsby-remark-copy-linked-files\`, \`gatsby-remark-smartypants\`]
    }
  }, \`gatsby-plugin-sharp\`, {
    resolve: \`gatsby-plugin-google-analytics\`,
    options: {
      trackingId: \`UA-774017-3\`
    }
  }, {
    resolve: \`gatsby-plugin-manifest\`,
    options: {
      name: \`Bricolage\`,
      short_name: \`Bricolage\`,
      icon: \`static/logo.png\`,
      start_url: \`/\`,
      background_color: redish,
      theme_color: redish,
      display: \`minimal-ui\`
    }
  }, \`gatsby-plugin-offline\`, // \`gatsby-plugin-preact\`,
  \`gatsby-plugin-react-helmet\`]
};",
  "describe": "Install gatsby-plugin-foo in gatsby-config.js",
  "diff": "[31m- Original  - 47[39m
[32m+ Modified  + 65[39m

[31m- const redish = \`#c5484d\`;[39m
[32m+ const redish = \`#c5484d\`[39m
[2m  module.exports = {[22m
[2m    siteMetadata: {[22m
[2m      title: \`Bricolage\`,[22m
[2m      author: \`Kyle Mathews\`,[22m
[31m-     homeCity: \`San Francisco\`[39m
[32m+     homeCity: \`San Francisco\`,[39m
[2m    },[22m
[31m-   plugins: [{[39m
[31m-     resolve: \`gatsby-source-filesystem\`,[39m
[31m-     options: {[39m
[31m-       path: \`\${__dirname}/src/pages\`,[39m
[31m-       name: \`pages\`[39m
[31m-     }[39m
[31m-   }, \`gatsby-transformer-sharp\`, \`gatsby-plugin-emotion\`, {[39m
[31m-     resolve: \`gatsby-plugin-typography\`,[39m
[31m-     options: {[39m
[31m-       pathToConfigModule: \`src/utils/typography\`[39m
[31m-     }[39m
[31m-   }, {[39m
[31m-     resolve: \`gatsby-transformer-remark\`,[39m
[31m-     options: {[39m
[31m-       plugins: [{[39m
[31m-         resolve: \`gatsby-remark-images\`,[39m
[31m-         options: {[39m
[31m-           maxWidth: 590[39m
[31m-         }[39m
[31m-       }, {[39m
[31m-         resolve: \`gatsby-remark-responsive-iframe\`,[39m
[31m-         options: {[39m
[31m-           wrapperStyle: \`margin-bottom: 1.0725rem\`[39m
[31m-         }[39m
[31m-       }, \`gatsby-remark-prismjs\`, \`gatsby-remark-copy-linked-files\`, \`gatsby-remark-smartypants\`][39m
[31m-     }[39m
[31m-   }, \`gatsby-plugin-sharp\`, {[39m
[31m-     resolve: \`gatsby-plugin-google-analytics\`,[39m
[31m-     options: {[39m
[31m-       trackingId: \`UA-774017-3\`[39m
[31m-     }[39m
[31m-   }, {[39m
[31m-     resolve: \`gatsby-plugin-manifest\`,[39m
[31m-     options: {[39m
[31m-       name: \`Bricolage\`,[39m
[31m-       short_name: \`Bricolage\`,[39m
[31m-       icon: \`static/logo.png\`,[39m
[31m-       start_url: \`/\`,[39m
[31m-       background_color: redish,[39m
[31m-       theme_color: redish,[39m
[31m-       display: \`minimal-ui\`[39m
[31m-     }[39m
[31m-   }, \`gatsby-plugin-offline\`, // \`gatsby-plugin-preact\`,[39m
[31m-   \`gatsby-plugin-react-helmet\`][39m
[31m- };[39m
[32m+   plugins: [[39m
[32m+     {[39m
[32m+       resolve: \`gatsby-source-filesystem\`,[39m
[32m+       options: {[39m
[32m+         path: \`\${__dirname}/src/pages\`,[39m
[32m+         name: \`pages\`,[39m
[32m+       },[39m
[32m+     },[39m
[32m+     \`gatsby-transformer-sharp\`,[39m
[32m+     \`gatsby-plugin-emotion\`,[39m
[32m+     {[39m
[32m+       resolve: \`gatsby-plugin-typography\`,[39m
[32m+       options: {[39m
[32m+         pathToConfigModule: \`src/utils/typography\`,[39m
[32m+       },[39m
[32m+     },[39m
[32m+     {[39m
[32m+       resolve: \`gatsby-transformer-remark\`,[39m
[32m+       options: {[39m
[32m+         plugins: [[39m
[32m+           {[39m
[32m+             resolve: \`gatsby-remark-images\`,[39m
[32m+             options: {[39m
[32m+               maxWidth: 590,[39m
[32m+             },[39m
[32m+           },[39m
[32m+           {[39m
[32m+             resolve: \`gatsby-remark-responsive-iframe\`,[39m
[32m+             options: {[39m
[32m+               wrapperStyle: \`margin-bottom: 1.0725rem\`,[39m
[32m+             },[39m
[32m+           },[39m
[32m+           \`gatsby-remark-prismjs\`,[39m
[32m+           \`gatsby-remark-copy-linked-files\`,[39m
[32m+           \`gatsby-remark-smartypants\`,[39m
[32m+         ],[39m
[32m+       },[39m
[32m+     },[39m
[32m+     \`gatsby-plugin-sharp\`,[39m
[32m+     {[39m
[32m+       resolve: \`gatsby-plugin-google-analytics\`,[39m
[32m+       options: {[39m
[32m+         trackingId: \`UA-774017-3\`,[39m
[32m+       },[39m
[32m+     },[39m
[32m+     {[39m
[32m+       resolve: \`gatsby-plugin-manifest\`,[39m
[32m+       options: {[39m
[32m+         name: \`Bricolage\`,[39m
[32m+         short_name: \`Bricolage\`,[39m
[32m+         icon: \`static/logo.png\`,[39m
[32m+         start_url: \`/\`,[39m
[32m+         background_color: redish,[39m
[32m+         theme_color: redish,[39m
[32m+         display: \`minimal-ui\`,[39m
[32m+       },[39m
[32m+     },[39m
[32m+     \`gatsby-plugin-offline\`, // \`gatsby-plugin-preact\`,[39m
[32m+     \`gatsby-plugin-react-helmet\`,[39m
[32m+     \\"gatsby-plugin-foo\\",[39m
[32m+   ],[39m
[32m+ }[39m
[32m+[39m",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "newState": "const redish = \`#c5484d\`
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`,
  },
  plugins: [
    {
      resolve: \`gatsby-source-filesystem\`,
      options: {
        path: \`\${__dirname}/src/pages\`,
        name: \`pages\`,
      },
    },
    \`gatsby-transformer-sharp\`,
    \`gatsby-plugin-emotion\`,
    {
      resolve: \`gatsby-plugin-typography\`,
      options: {
        pathToConfigModule: \`src/utils/typography\`,
      },
    },
    {
      resolve: \`gatsby-transformer-remark\`,
      options: {
        plugins: [
          {
            resolve: \`gatsby-remark-images\`,
            options: {
              maxWidth: 590,
            },
          },
          {
            resolve: \`gatsby-remark-responsive-iframe\`,
            options: {
              wrapperStyle: \`margin-bottom: 1.0725rem\`,
            },
          },
          \`gatsby-remark-prismjs\`,
          \`gatsby-remark-copy-linked-files\`,
          \`gatsby-remark-smartypants\`,
        ],
      },
    },
    \`gatsby-plugin-sharp\`,
    {
      resolve: \`gatsby-plugin-google-analytics\`,
      options: {
        trackingId: \`UA-774017-3\`,
      },
    },
    {
      resolve: \`gatsby-plugin-manifest\`,
      options: {
        name: \`Bricolage\`,
        short_name: \`Bricolage\`,
        icon: \`static/logo.png\`,
        start_url: \`/\`,
        background_color: redish,
        theme_color: redish,
        display: \`minimal-ui\`,
      },
    },
    \`gatsby-plugin-offline\`, // \`gatsby-plugin-preact\`,
    \`gatsby-plugin-react-helmet\`,
    \\"gatsby-plugin-foo\\",
  ],
}
",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin destroy 1`] = `undefined`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin update 1`] = `
Object {
  "_message": "Installed gatsby-plugin-foo in gatsby-config.js",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin update plan 1`] = `
Object {
  "currentState": "const redish = \`#c5484d\`
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`,
  },
  plugins: [
    {
      resolve: \`gatsby-source-filesystem\`,
      options: {
        path: \`\${__dirname}/src/pages\`,
        name: \`pages\`,
      },
    },
    \`gatsby-transformer-sharp\`,
    \`gatsby-plugin-emotion\`,
    {
      resolve: \`gatsby-plugin-typography\`,
      options: {
        pathToConfigModule: \`src/utils/typography\`,
      },
    },
    {
      resolve: \`gatsby-transformer-remark\`,
      options: {
        plugins: [
          {
            resolve: \`gatsby-remark-images\`,
            options: {
              maxWidth: 590,
            },
          },
          {
            resolve: \`gatsby-remark-responsive-iframe\`,
            options: {
              wrapperStyle: \`margin-bottom: 1.0725rem\`,
            },
          },
          \`gatsby-remark-prismjs\`,
          \`gatsby-remark-copy-linked-files\`,
          \`gatsby-remark-smartypants\`,
        ],
      },
    },
    \`gatsby-plugin-sharp\`,
    {
      resolve: \`gatsby-plugin-google-analytics\`,
      options: {
        trackingId: \`UA-774017-3\`,
      },
    },
    {
      resolve: \`gatsby-plugin-manifest\`,
      options: {
        name: \`Bricolage\`,
        short_name: \`Bricolage\`,
        icon: \`static/logo.png\`,
        start_url: \`/\`,
        background_color: redish,
        theme_color: redish,
        display: \`minimal-ui\`,
      },
    },
    \`gatsby-plugin-offline\`, // \`gatsby-plugin-preact\`,
    \`gatsby-plugin-react-helmet\`,
    \\"gatsby-plugin-foo\\",
  ],
}
",
  "describe": "Install gatsby-plugin-foo in gatsby-config.js",
  "diff": "[2mCompared values have no visual difference.[22m",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "newState": "const redish = \`#c5484d\`
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`,
  },
  plugins: [
    {
      resolve: \`gatsby-source-filesystem\`,
      options: {
        path: \`\${__dirname}/src/pages\`,
        name: \`pages\`,
      },
    },
    \`gatsby-transformer-sharp\`,
    \`gatsby-plugin-emotion\`,
    {
      resolve: \`gatsby-plugin-typography\`,
      options: {
        pathToConfigModule: \`src/utils/typography\`,
      },
    },
    {
      resolve: \`gatsby-transformer-remark\`,
      options: {
        plugins: [
          {
            resolve: \`gatsby-remark-images\`,
            options: {
              maxWidth: 590,
            },
          },
          {
            resolve: \`gatsby-remark-responsive-iframe\`,
            options: {
              wrapperStyle: \`margin-bottom: 1.0725rem\`,
            },
          },
          \`gatsby-remark-prismjs\`,
          \`gatsby-remark-copy-linked-files\`,
          \`gatsby-remark-smartypants\`,
        ],
      },
    },
    \`gatsby-plugin-sharp\`,
    {
      resolve: \`gatsby-plugin-google-analytics\`,
      options: {
        trackingId: \`UA-774017-3\`,
      },
    },
    {
      resolve: \`gatsby-plugin-manifest\`,
      options: {
        name: \`Bricolage\`,
        short_name: \`Bricolage\`,
        icon: \`static/logo.png\`,
        start_url: \`/\`,
        background_color: redish,
        theme_color: redish,
        display: \`minimal-ui\`,
      },
    },
    \`gatsby-plugin-offline\`, // \`gatsby-plugin-preact\`,
    \`gatsby-plugin-react-helmet\`,
    \\"gatsby-plugin-foo\\",
  ],
}
",
}
`;
